src/askui/models/askui/api.py

```diff
@@ -4,41 +4,73 @@
 import requests
 
 from PIL import Image
-from typing import Union
+from typing import List, Union
+from askui.models.askui.ai_element_utils import AiElement, AiElementCollection
 from askui.utils import image_to_base64
 from askui.logging import logger
 
 
+
 class AskUIHandler:
     def __init__(self):
         self.inference_endpoint = os.getenv("ASKUI_INFERENCE_ENDPOINT", "https://inference.askui.com")
         self.workspace_id = os.getenv("ASKUI_WORKSPACE_ID")
         self.token = os.getenv("ASKUI_TOKEN")
-
+    
         self.authenticated = True
         if self.workspace_id is None or self.token is None:
             logger.warning("ASKUI_WORKSPACE_ID or ASKUI_TOKEN missing.")
             self.authenticated = False
 
+        self.ai_element_collection = AiElementCollection()
+
+
+
     def __build_askui_token_auth_header(self, bearer_token: str | None = None) -> dict[str, str]:
         if bearer_token is not None:
             return {"Authorization": f"Bearer {bearer_token}"}
         token_base64 = base64.b64encode(self.token.encode("utf-8")).decode("utf-8")
         return {"Authorization": f"Basic {token_base64}"}
     
+    def __build_custom_elements(self, ai_elements: List[AiElement]):
+        """
+        Converts AiElements to the CustomElementDto format expected by the backend.
+        
+        Args:
+            ai_elements (List[AiElement]): List of AI elements to convert
+            
+        Returns:
+            dict: Custom elements in the format expected by the backend
+        """
+        if not ai_elements:
+            return {}
+        
+        custom_elements = []
+        for element in ai_elements:
+            custom_element = {
+                "customImage": "," + image_to_base64(element.image),            
+                "imageCompareFormat": "grayscale",
+                "name": element.metadata.name
+            }
+            custom_elements.append(custom_element)
+        
+        return {
+            "customElements": custom_elements
+        }  
```

Comment:
Good places for `TypedDict` (https://peps.python.org/pep-0589/)

Reply:
I would skip this for now. Can be refactored later